### ISSUE TITLE
Variable is already defined in class

### ISSUE THREAD

**Issue by TrekSoft** [aws-amplify/amplify-cli#5496]
If a user puts a property in their GraphQL schema in all caps the Java code generation creates a conflict between the ModelField variable name and the QueryField variable name because they are both named identically. This would be a breaking change to fix so perhaps we need to just document that we don't support all caps type fields but wanted to raise the issue here to discuss.

**Comment by richardmcclellan** [comment]
Similar issue: https://github.com/aws-amplify/amplify-cli/issues/5046

**Comment by TrekSoft** [comment]
Yeah that's a different root cause than this one. Those others look like they can be fixed without a breaking change. This one most likely needs to be addressed with a documentation note forbidding the use of all caps fields and a CLI check when compiling the schema to catch if the user has an all caps type and forbid it.

**Comment by AaronZyLee** [comment]
Currently we use the exact field name from graphql schema as the `ModelField` variable name while use a `constantCase` for `QueryField` ones, which is a case of all caps letters. That's why they are named identically. This can be fixed by adding a regulation on the `ModelField` name, which can be either lowercase or camelCase. Or if there are any breaking changes we can simply adjust the documentations.

**Comment by TrekSoft** [comment]
Yeah I think the best solution for now is to put a note in the documentation saying the fields in a type cannot be all caps and then have the CLI check for that when they go to push and fail with a validation error if they have an all caps field in a type.
